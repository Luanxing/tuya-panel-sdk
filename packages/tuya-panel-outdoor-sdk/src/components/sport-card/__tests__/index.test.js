import React from 'react';
import { shallow } from 'enzyme';
import SportCard from '../index';

describe('sport-card components', () => {
  it('basic render', () => {
    const cardDatas = [
      {
        dp: 'activetime_day',
        bgColor: '#24D671',
        iconFont: '',
        unit: 'h',
        tip: '',
        value: '',
      },
      {
        dp: 'calories_day',
        bgColor: '#FE9456',
        iconFont: '',
        unit: 'cal',
        tip: '',
        value: '',
      },
      {
        dp: 'distance_day',
        bgColor: '#F6BA7B',
        iconFont: '',
        unit: 'km',
        tip: '',
        value: '',
      },
    ];
    const wrapper = shallow(<SportCard dpDatas={cardDatas} />);
    expect(wrapper).toMatchSnapshot();
  });
});
