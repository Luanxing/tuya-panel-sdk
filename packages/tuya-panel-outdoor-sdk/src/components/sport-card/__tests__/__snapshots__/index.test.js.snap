// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sport-card components basic render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 100,
    }
  }
>
  <View
    key="activetime_day"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "baseline",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 44,
              "fontWeight": "bold",
            },
            Object {
              "color": "#000",
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontSize": 28,
            },
            Object {
              "color": "#000",
            },
          ]
        }
      >
        h
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 14,
              "height": 28,
              "marginRight": 12,
              "width": 28,
            },
            Object {
              "backgroundColor": "#24D671",
            },
          ]
        }
      >
        <Component
          color="#fff"
          d=""
          size={28}
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "lineHeight": 28,
            },
            Object {
              "color": "#000",
              "opacity": 1,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    key="calories_day"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "baseline",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 44,
              "fontWeight": "bold",
            },
            Object {
              "color": "#000",
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontSize": 28,
            },
            Object {
              "color": "#000",
            },
          ]
        }
      >
        cal
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 14,
              "height": 28,
              "marginRight": 12,
              "width": 28,
            },
            Object {
              "backgroundColor": "#FE9456",
            },
          ]
        }
      >
        <Component
          color="#fff"
          d=""
          size={28}
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "lineHeight": 28,
            },
            Object {
              "color": "#000",
              "opacity": 1,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    key="distance_day"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "baseline",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 44,
              "fontWeight": "bold",
            },
            Object {
              "color": "#000",
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontSize": 28,
            },
            Object {
              "color": "#000",
            },
          ]
        }
      >
        km
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 14,
              "height": 28,
              "marginRight": 12,
              "width": 28,
            },
            Object {
              "backgroundColor": "#F6BA7B",
            },
          ]
        }
      >
        <Component
          color="#fff"
          d=""
          size={28}
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "lineHeight": 28,
            },
            Object {
              "color": "#000",
              "opacity": 1,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
